juice/src/layer.rs

```diff
@@ -537,10 +537,22 @@ impl<B: IBackend> Layer<B> {
             // reshape input tensor to the reshaped shape
             let old_shape = self.input_blobs_data[input_i].read().unwrap().desc().clone();
             if old_shape.size() != reshaped_shape.size() {
-                panic!(
-                    "Input Shape Mismatch\nExpected {:?}\nActual {:?}",
-                    reshaped_shape, old_shape
-                );
+                if reshaped_shape[1..] == old_shape[1..] {
```

Comment:
I think it'd make more sense to add an assert per dimension, and just fail with the first.

```rs
for dim in 0..5 {
     let old = old_shape[dim];
     let reshaped = reshaped__shape[dim];
     assert_eq!(old, reshpaed, "Dimension {dim} mismatches {old} != {reshaped}"); 
}
unreachable!("If the total size is not equal, one dimension mismatch must exist. qed");
```

Reply:
I did it this way to catch some errors with mismatched batch sizes to make it a bit more clear.

Reply:
I believe that my recent commit should have done this.